### PR TITLE
perf(metadata): intern positions once per location, not once per record (#226)

### LINKED ISSUE #226: Positions are stored as formatted strings: 1GB (13.5%) of allocation, and the cause of #216
Positions are stored as **formatted strings** (`getPosition` → `fset.Position(pos).String()`), one per recorded node. On a 360-package project that is **1.03GB — 13.5% of all allocation** (`go/token.Position.String`), second only to type rendering (now memoized) and `NewCallArgument` itself.

The cost is structural: every `CallArgument`, `Call`, function, assignment and struct instance formats `/abs/path/file.go:12:34` into a fresh string, and those strings are then interned into the pool anyway. The formatting happens first, so interning does not save the allocation.

It also causes a correctness problem already filed separately: because `fset.Position(...).String()` renders the **absolute** path, a closure handler's `operationId` embeds the local filesystem path, so the same code produces different specs on different machines (#216). One change fixes both.

## Directions

1. **Store the components, format at the boundary.** Keep `(file string-pool index, line, col int)` — three ints, no allocation — and render only where a position reaches output (diagnostics, the `FuncLit` naming path). This is the same discipline the type model already follows for types (`docs/TYPE_MODEL.md`: build a structured value, render at an output boundary, never parse the string back).
2. **Intern before formatting.** Cheaper to implement: format into a reusable buffer and intern, so the allocation is per *distinct* position rather than per recorded node. Positions repeat heavily (one per node in a file), so this alone should remove most of the GB — but it keeps the absolute path, so it does not address #216.
3. Whichever is chosen, make the file component **module-relative** to fix #216 at the same time.

## Constraints

Positions are load-bearing in more places than they look:

- they are part of instance IDs (`CallIdentifier`), so the *format* is observable in tracker keys and in `operationId`s for closures;
- they are byte-compared in the metadata goldens (`internal/spec/tests/*.yaml`), so a representation change is a golden regeneration (`-update`), which must be a deliberate, reviewed step rather than a side effect;
- `docs/DEBUGGING.md` and the diagram server surface them to users.

So this is the largest-payoff remaining allocation win, but it is not a local change — it wants its own PR, with the golden regeneration isolated in its own commit.

## Acceptance

- `go/token.Position.String` no longer appears among the top allocators on a large real project.
- Spec output byte-identical across two checkouts at **different paths** (that is #216's acceptance too).
- Metadata goldens regenerated in a separate, reviewable commit.

### PR DESCRIPTION
**Pull request by ehabterra**
Closes #226.

## The problem

`getPosition` called `go/token.Position.String` for **every recorded node** — every `CallArgument`, `Call`, function, assignment and struct instance — and the result was interned afterwards, so interning saved nothing. The string was already allocated by then.

On gitea (360 packages) that was **1012MB of a 4766MB run — 21% of all allocation**, second only to `NewCallArgument`.

## The fix

The same source location is recorded many times: once per node that mentions it, and again each time a node is revisited. So the pool index is memoized per `token.Pos` — a byte offset within the `FileSet`, which identifies a location *exactly*, so the cache cannot collide two call sites — and a repeat costs one map lookup.

`Position.String` is reproduced by hand rather than called, because it builds its result with `fmt`.

## Why the hand-rolled formatter is safe

That rendering is observable in tracker keys, in a closure's identity (`FuncLit:<position>`) and in the byte-compared metadata goldens. `TestFormatPositionMatchesToken` checks every shape `token` treats specially — no column, no filename, invalid position, neither — **against `token` itself**, so the two cannot drift silently.

## Measured (gitea, 360 packages)

| | before | after |
|---|---|---|
| total `alloc_space` | 4766 MB | **3805 MB** (−20%) |
| `token.Position.String` | 1012 MB (21.2%) | **absent** |
| position formatting | 1035 MB cum | 72 MB (−93%) |

Wall clock is unchanged **within this machine's noise** — the same binary varied 52–85s across runs — so the win claimed here is allocation and GC pressure, not time. The metadata stage averaged ~10% faster over three interleaved pairs, with run-to-run spread of the same order; not claimed as a result.

## Verification

- Output **byte-identical** on gitea and across all 67 fixtures.
- **Goldens needed no regeneration** (the issue expected one): they were already written with relative filenames, and the rendering is unchanged.
- Full suite, `go test -race`, `go vet`, `gofmt`, `golangci-lint` (0 issues); coverage 96.0% against the 96% floor.

## Incidental

The three sites that mint a closure's `FuncLit:<position>` name now share `funcLitName`, so the shape is stated once — that name is what #216 (non-reproducible operationIds) is about, and it is the next PR.

**Found while verifying this, unrelated to it:** the unmodified binary produces *different specs on consecutive runs* on photoprism — an enum inferred from constants of a `type X = string` **alias**, where two aliases in one package are the same type, so the group is picked by map order. Filed separately; half the runs document the wrong enum.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **Improvements**
  - Improved source-position tracking across generated metadata, including functions, variables, assignments, calls, and interface resolutions.
  - Added consistent naming for anonymous functions, making them easier to identify.
  - Reduced repeated position processing to improve metadata generation efficiency and consistency.

- **Bug Fixes**
  - Corrected handling of missing or invalid source positions.
  - Improved position handling for generic expressions and nested call analysis.

- **Tests**
  - Added coverage for position formatting, caching, anonymous-function naming, and edge cases.

### CODE REVIEW

File: internal/metadata/position.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	}
+
+	m.posMutex.RLock()
+	idx, ok := m.posCache[pos]
+	m.posMutex.RUnlock()
+	if ok {
+		return idx
+	}
+
+	idx = m.StringPool.Get(formatPosition(fset.Position(pos)))
```
[the remark this answers is not in the document]
Comment (ehabterra) [reply internal/metadata/position.go]: Correct as a fact, deliberately out of scope here: that is #216, which is its own PR by request (one issue per PR).

This PR is a pure allocation change whose acceptance is **byte-identical output** — verified on gitea and across all 67 fixtures. Normalizing filenames changes every closure's `FuncLit:<position>` name, and with it tracker keys and operationIds, so folding it in would conflate a measurable perf change with a behaviour change and make both harder to review or revert.

Groundwork for it is in this PR on purpose: the three sites that mint a closure name now share `funcLitName`, and all position rendering goes through one place, so #216 becomes a small change on top rather than a sweep.
